src/fungible_token.rs

```diff
@@ -0,0 +1,452 @@
+                    #[cfg(feature = "log")]
+                    sdk::log(&format!(
+                        "Refund amount {} from {} to {}",
+                        refund_amount, receiver_id, sender_id
+                    ));
+                    (amount - refund_amount, 0)
+                } else {
+                    // Sender's account was deleted, so we need to burn tokens.
+                    self.total_supply -= refund_amount;
+                    #[cfg(feature = "log")]
+                    sdk::log("The account of the sender was deleted");
+                    (amount, refund_amount)
+                };
+            }
+        }
+        (amount, 0)
+    }
+
+    pub fn ft_resolve_transfer(
+        &mut self,
+        sender_id: &str,
+        receiver_id: &str,
+        amount: u128,
+    ) -> u128 {
+        self.internal_ft_resolve_transfer(sender_id, receiver_id, amount)
+            .0
+    }
+
+    pub fn internal_storage_unregister(
+        &mut self,
+        force: Option<bool>,
+    ) -> Option<(AccountId, Balance)> {
+        sdk::assert_one_yocto();
+        let account_id_key = sdk::predecessor_account_id();
+        let account_id = str_from_slice(&account_id_key);
+        let force = force.unwrap_or(false);
+        if let Some(balance) = self.accounts_get(account_id) {
+            let balance = u128::try_from_slice(&balance[..]).unwrap();
+            if balance == 0 || force {
+                self.accounts_remove(account_id);
+                self.total_supply -= balance;
+                let amount = self.storage_balance_bounds().min + 1;
+                let promise0 = sdk::promise_batch_create(&account_id_key);
+                sdk::promise_batch_action_transfer(promise0, amount);
+                Some((account_id.to_string(), balance))
+            } else {
+                sdk::panic_utf8(b"ERR_FAILED_UNREGISTER_ACCOUNT_POSITIVE_BALANCE")
+            }
+        } else {
+            #[cfg(feature = "log")]
+            sdk::log(&format!("The account {} is not registered", &account_id));
+            None
+        }
+    }
+
+    pub fn storage_balance_bounds(&self) -> StorageBalanceBounds {
+        let required_storage_balance =
+            Balance::from(self.account_storage_usage) * sdk::storage_byte_cost();
+        StorageBalanceBounds {
+            min: required_storage_balance,
+            max: Some(required_storage_balance),
+        }
+    }
+
+    pub fn internal_storage_balance_of(&self, account_id: &str) -> Option<StorageBalance> {
+        if self.accounts_contains_key(account_id) {
+            Some(StorageBalance {
+                total: self.storage_balance_bounds().min,
+                available: 0,
+            })
+        } else {
+            None
+        }
+    }
+
+    pub fn storage_balance_of(&self, account_id: &str) -> Option<StorageBalance> {
+        self.internal_storage_balance_of(account_id)
```

Comment:
Return just `StorageBalance` instead of `Option` (see above).

Reply:
Fixed at #98